### ISSUE TITLE
[BUG] Configuration Build Failed

### ISSUE THREAD

**Issue by MeCJay12** [WillyXJ/facileManager#437]
Please prefix your issue title with one of the following: [BUG], [ISSUE], [FEATURE REQUEST], [MODULE REQUEST], [OTHER].

Replace everything between stars with current version of your facileManager and module installations:  
fM Version : **3.4**
{Module Name} Version : **3.3.1**

**In raising this issue, I confirm the following (please check boxes, eg [X]):**

- [X] I have read and understood the [contributors guide](https://github.com/WillyXJ/facileManager/blob/master/.github/CONTRIBUTING.md).
- [X] I have checked that the bug-fix I am reporting can be replicated, or that the feature I am suggesting isn't already present.
- [X] I have checked that the issue I'm posting isn't already reported.
- [X] I have checked that the issue I'm posting isn't already solved and no duplicates exist in [closed issues](https://github.com/WillyXJ/facileManager/issues?q=is%3Aissue+is%3Aclosed) and [opened issues](https://github.com/WillyXJ/facileManager/issues)
- [X] I have checked the pull requests tab for existing solutions/implementations to my issue/suggestion.

---

**(BUG | ISSUE) Expected Behavior:**
When clicking the push updates button in the top bar, updates would be pushed to servers. Also, in the servers menu, when clicking the wrench next to a server that the config for the server would be built. This appears to be a results of updating to 3.4 and 3.3.1. Found [this](https://github.com/WillyXJ/facileManager/issues/6) old post but the information seems to be out of date.

**(BUG | ISSUE) Actual Behavior:**
"Configuration build failure". Old name servers started doing the  same.

**(BUG | ISSUE) Steps to reproduce:**
Attempt to push update or build config

**Comment by WillyXJ** [comment]
What errors do you get when you run the configuration build manually from your DNS server?

```
sudo php /usr/local/facileManager/fmDNS/client.php buildconf debug
```

**Comment by MeCJay12** [comment]
That's the command I was looking for!

So when I run that command I get:
`Server is not found.`

I re-ran the install script and it connected to the server fine. The server showed up in the WebUI. I enable the server and make my changes (I use IP rather than hostname, run-as bind, and /var/lib/bind/zones instead of /etc/bind/zones). When I go to push the config from the server I get the same error.

If I run the command right after the install script, I get:
`DNS server is disabled.`
which makes sense. Even without making any changes, other than IP, I still get failed to build config.

**Comment by WillyXJ** [comment]
For some reason your client is not found in the database when querying based on serial number, but it is found when the client is disabled.  If it's found in the latter scenario then it should be found always. 
 Any chance you can post the contents of /usr/local/facileManager/config.inc.php or email it (support at facilemanager ..dot.. com)?  Also, what are your fM server and fmDNS client environments (OS, PHP, MySQL versions)?

**Comment by MeCJay12** [comment]
/usr/local/facileManager/config.inc.php:
`<?php

/**
 * Contains configuration details for facileManager
 *
 * @package facileManager
 *
 */

/** facileManager server */
define('FMHOST', 'Redacted-IP/facileManager/');

/** Account number */
define('AUTHKEY', 'default');

/** Server unique serial number */
define('SERIALNO', '225775909');`

The environment is all Ubuntu Server 16.04, PHP 7.0.33-0ubuntu0.16.04.2, and (on the server) MySQL Ver 14.14 Distrib 5.7.25.

**Comment by WillyXJ** [comment]
If you run the following SQL query on your database, do you get a valid result that includes your DNS server?

```
SELECT * FROM `fm_dns_servers` WHERE `server_status`!='deleted' AND `account_id`='1' AND `server_serial_no`='225775909' AND server_installed='yes';
```

**Comment by MeCJay12** [comment]
mysql> use facileManager
Database changed
mysql> SELECT * FROM "fm_dns_servers" WHERE "server_status"!='deleted' AND "account_id"='1' AND "server_serial_no"='225775909' AND server_installed='yes';
Empty set (0.00 sec)

UI at the time the command was run
![Capture](https://user-images.githubusercontent.com/43658705/54690042-f7b8bb80-4af6-11e9-8aca-147092ffa578.PNG)

**Comment by WillyXJ** [comment]
Interesting.  So there seems to be some kind of mismatch in the data.  What do you get for this query?

```
SELECT server_name,server_serial_no,server_status,server_installed,account_id FROM `fm_dns_servers`;
```
Ideally, the server_serial_no needs to match what's in config.inc.php on the DNS servers.

**Comment by MeCJay12** [comment]
mysql> SELECT server_name,server_serial_no,server_status,server_installed,account_id FROM `fm_dns_servers`;
+----------------+------------------+---------------+------------------+------------+
| server_name    | server_serial_no | server_status | server_installed | account_id |
+----------------+------------------+---------------+------------------+------------+
| IP-Prefix.75 |        679942854 | active        | yes              |          1 |
| IP-Prefix.78 |        852573034 | active        | yes              |          1 |
| IP-Prefix.77 |        187136358 | active        | yes              |          1 |
| IP-Prefix.77 |        831863742 | deleted       | yes              |          1 |
| IP-Prefix.78 |        343920963 | deleted       | yes              |          1 |
| IP-Prefix.77 |        965106748 | deleted       | yes              |          1 |
| IP-Prefix.75 |        249826481 | deleted       | yes              |          1 |
+----------------+------------------+---------------+------------------+------------+
7 rows in set (0.00 sec)

Interesting. The serial above doesn't match any of these.

**Comment by WillyXJ** [comment]
I don't see the serial number in that output that is from your config.inc.php file.  If you update your config.inc.php file on the DNS servers with their respective serial numbers from the database query (for the active results), this issue should become resolved.  You can either manually change that file or remove it and run through the installer again.  It does make me wonder how the serial numbers got mismatched in the first place though.

**Comment by MeCJay12** [comment]
You are right. If I take the serials from the query and load them into the DNS servers it work again. It looks like when I first install the DNS servers the serial is correct then when I change the server from its reported hostname to its IP the serial gets changed in the server.

**Comment by WillyXJ** [comment]
Great! I've also confirmed the serial number changes after changing the hostname which is indeed unexpected behavior.  A fix will be coming in the next release.

**Comment by WillyXJ** [comment]
This is now fixed in [fmDNS 3.3.2](http://www.facilemanager.com/modules/fmdns/) and later.
